### PR TITLE
Move site deployments to use netlify-cli

### PR DESCRIPTION
**Pull request by mattcompiles**
I have already disabled builds in the Netlify dashboard so this is safe to merge. 

Check out a draft deploy here: https://633146a89dc9973468f7c5f4--vanilla-extract-css.netlify.app

### CODE REVIEW

File: .github/workflows/deploy-site.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+jobs:
+  deploy-site:
+    name: Deploy site
+    runs-on: ubuntu-latest
+    env:
+      CI: true
+    steps:
+      - uses: pnpm/action-setup@v2.0.1
+        with:
+          version: 7.9.5
```
Comment (mrm007) [inline .github/workflows/deploy-site.yml:14]: `version` is no longer required as of `2.2.2`

https://github.com/pnpm/action-setup#version

```suggestion
      - uses: pnpm/action-setup@v2.2.2
```

File: .github/workflows/deploy-site.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+      - name: Checkout Repo
+        uses: actions/checkout@main
+
+      - name: Setup Node.js
```
Comment (mrm007) [inline .github/workflows/deploy-site.yml:19]: Typo:

```suggestion
      - name: Set up Node.js
```

**Review by mrm007 — APPROVED** [review]

File: .github/workflows/deploy-site.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
+name: Deploy site
+
+on: push
```
Comment (mrm007) [inline .github/workflows/deploy-site.yml:3]: This won’t trigger for pull requests from non-maintainers i.e. from forked repos. 

```suggestion
on:
  pull_request:
  push:
    branches: [master]
```

Comment (mattcompiles) [reply .github/workflows/deploy-site.yml:3]: This was kind of intentional. Firstly, it's nice to deploy the site without having to make a PR. I was on the fence about external PRs, however external PRs rarely ever need to review site changes. Could change over time though.

**Pull request merged**